examples/onnxruntime/quantization/image-classification/run_image_classification.py

```diff
@@ -0,0 +1,434 @@
+        ]
+        return example_batch
+
+    metric = load_metric("accuracy")
+
+    # You can define your custom compute_metrics function. It takes an `EvalPrediction` object (a namedtuple with a
+    # predictions and label_ids field) and has to return a dictionary string to float.
+    def compute_metrics(p: EvalPrediction):
+        preds = p.predictions[0] if isinstance(p.predictions, tuple) else p.predictions
+        preds = np.argmax(preds, axis=1)
+
+        result = metric.compute(predictions=preds, references=p.label_ids)
+        return result
+
+    # Create the quantizer
+    quantizer = ORTQuantizer(
+        preprocessor=feature_extractor, model=model, feature="image-classification", opset=optim_args.opset
+    )
+
+    apply_static_quantization = optim_args.quantization_approach == "static"
+
+    # Create the quantization configuration containing all the quantization parameters
+    qconfig = QuantizationConfig(
+        is_static=apply_static_quantization,
+        format=QuantFormat.QDQ if apply_static_quantization else QuantFormat.QOperator,
+        mode=QuantizationMode.QLinearOps if apply_static_quantization else QuantizationMode.IntegerOps,
+        activations_dtype=QuantType.QInt8 if apply_static_quantization else QuantType.QUInt8,
+        weights_dtype=QuantType.QInt8,
+        per_channel=optim_args.per_channel,
+        reduce_range=optim_args.reduce_range,
+        operators_to_quantize=["MatMul", "Add"],
+    )
+
+    ranges = None
+    # Create a quantization preprocessor to determine the nodes to exclude
+    quantization_preprocessor = QuantizationPreprocessor()
+    if apply_static_quantization:
+        calibration_dataset = dataset["train"]
+        if optim_args.num_calibration_samples is not None:
+            calibration_dataset = calibration_dataset.select(range(optim_args.num_calibration_samples))
+
+        # all images are loaded in memory, which could prove expensive if num_calibration_samples is large
+        calibration_dataset = calibration_dataset.map(
+            partial(preprocess_function),
+            batched=True,
+            load_from_cache_file=not data_args.overwrite_cache,
+            desc="Running preprocessing on calibration dataset",
+        )
+
+        # Remove the unnecessary columns of the calibration dataset before the calibration step
+        calibration_dataset = quantizer.clean_calibration_dataset(calibration_dataset)
+
+        # Create the calibration configuration given the selected calibration method
+        if optim_args.calibration_method == "entropy":
+            calibration_config = AutoCalibrationConfig.entropy(calibration_dataset)
+        elif optim_args.calibration_method == "percentile":
+            calibration_config = AutoCalibrationConfig.percentiles(
+                calibration_dataset,
+                percentile=optim_args.calibration_histogram_percentile,
+            )
```

Comment:
Depending on the operators we wish to quantize, entropy and percentile calibration may return an error on ORT's side. Just to be sure, do the 3 calibration methods work with the calibration config `qconfig` defined above?

Reply:
Indeed, we do get an error for percentile (at least):

```
Collecting tensor data and making histogram ...
/home/fxmarty/anaconda3/envs/hf-inf/lib/python3.9/site-packages/onnxruntime/quantization/calibrate.py:539: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.
  data_arr = np.asarray(data_arr)
Traceback (most recent call last):
  File "/home/fxmarty/hf_internship/optimum/examples/onnxruntime/quantization/image-classification/run_image_classification.py", line 432, in <module>
    main()
  File "/home/fxmarty/hf_internship/optimum/examples/onnxruntime/quantization/image-classification/run_image_classification.py", line 369, in main
    quantizer.partial_fit(
  File "/home/fxmarty/hf_internship/optimum/optimum/onnxruntime/quantization.py", line 260, in partial_fit
    self._calibrator.collect_data(reader)
  File "/home/fxmarty/anaconda3/envs/hf-inf/lib/python3.9/site-packages/onnxruntime/quantization/calibrate.py", line 391, in collect_data
    self.collector.collect(clean_merged_dict)
  File "/home/fxmarty/anaconda3/envs/hf-inf/lib/python3.9/site-packages/onnxruntime/quantization/calibrate.py", line 502, in collect
    return self.collect_value(name_to_arr)
  File "/home/fxmarty/anaconda3/envs/hf-inf/lib/python3.9/site-packages/onnxruntime/quantization/calibrate.py", line 543, in collect_value
    min_value = np.min(data_arr)
  File "<__array_function__ internals>", line 5, in amin
  File "/home/fxmarty/anaconda3/envs/hf-inf/lib/python3.9/site-packages/numpy/core/fromnumeric.py", line 2879, in amin
    return _wrapreduction(a, np.minimum, 'min', axis, None, out,
  File "/home/fxmarty/anaconda3/envs/hf-inf/lib/python3.9/site-packages/numpy/core/fromnumeric.py", line 86, in _wrapreduction
    return ufunc.reduce(obj, axis, dtype, out, **passkwargs)
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```

Most likely due to the warning when exporting to onnx ( https://github.com/huggingface/transformers/blob/401fcca6c561d61db6ce25d9b1cebb75325a034f/src/transformers/models/vit/modeling_vit.py#L171 )
```
/home/fxmarty/hf_internship/transformers/src/transformers/models/vit/modeling_vit.py:187: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  if height != self.image_size[0] or width != self.image_size[1]:
```

Reply:
Ok for me to add a check for `image-*` task and calibration methods to print out a message informing the user.